Go Live version

davidwaroquiers opened the pull request:
First production version.

gpetretto committed:
First minimal working implementation

gpetretto committed:
add submit function

gpetretto committed:
revised configurations

gpetretto committed:
first daemon implementation based on supervisor

gpetretto committed:
refactor modules, JobController and CLI

davidwaroquiers committed:
Added dependencies in pyproject.toml.

gpetretto committed:
updates on queries and CLI

gpetretto committed:
move remote document inside Firework document

gpetretto committed:
more functionalities and bug fixes

gpetretto committed:
improve check_run_status and more CLI

gpetretto committed:
helper to generate configuration

gpetretto committed:
more refactoring of configuration

gpetretto committed:
refactor project cli

gpetretto committed:
Merge remote-tracking branch 'origin/config_deps' into worker

gpetretto committed:
more CLI options

gpetretto committed:
Merge pull request #1 from Matgenix/config

gpetretto committed:
refactor exec config and typer error handling

gpetretto committed:
Merge branch 'config' into develop

gpetretto committed:
more CLI. fix FW task

gpetretto committed:
update remote host

gpetretto committed:
improve cli

gpetretto committed:
optional warnings in ConfigManager

gpetretto committed:
document configuration options and revise RemoteWorker

gpetretto committed:
get_jobstore function

ml-evs committed:
Temporarily add qtoolkit as git dependency

ml-evs committed:
Use default/first worker if not passed to submit function

ml-evs committed:
Add simple usage instructions (need checking)

ml-evs committed:
Use strict jobflow deps to avoid pydantic errors

gpetretto committed:
Add .DS_store to gitignore

gpetretto committed:
enforce connecting of the remote host

ml-evs committed:
Pin to the development jobflow

ml-evs committed:
Remove incorrect usage docs

gpetretto committed:
restart connection for remote host if connection is dropped

ml-evs committed:
Enable `ConfigManager` warnings for `jf project check`

gpetretto committed:
add run time information

ml-evs committed:
Handle case that date in fw dict is already a date

ml-evs committed:
Use string datetime when pinging jobs

davidwaroquiers committed:
Merge pull request #7 from ml-evs/develop

A few minor tweaks from first use

davidwaroquiers committed:
Added documentation structure.

gpetretto committed:
cli message for global options

gpetretto committed:
fix FlowInfo for string updated_on

ml-evs committed:
Move fallback worker config up to submit function

ml-evs committed:
Do not allow null workers when constructing a firework

gpetretto committed:
Merge pull request #12 from ml-evs/ml-evs/fix

Fix handling of null worker in `submit_flow`

gpetretto committed:
improve CLI, include allow_external_references

ml-evs committed:
Add work_dir checker to project check CLI

- Delete canary file after use in project check

ml-evs committed:
Use `Path` for remote operations to deal with spaces in dir names

ml-evs committed:
Add validator for absoluteness of `work_dir`

gpetretto committed:
disable job_ids options in the CLI

ml-evs committed:
Format `error_job` the same way as `error_remote`

gpetretto committed:
Merge pull request #15 from ml-evs/ml-evs/fix

Check `work_dir` with project check CLI and enforce absolute paths

gpetretto committed:
restore queries based on job id

davidwaroquiers committed:
Fixed jf flow list.

davidwaroquiers committed:
Fixed state query in jobcontroller.

gpetretto committed:
more fix for id based searches

gpetretto committed:
Merge pull request #17 from Matgenix/job_id

Restore queries based on job id

gpetretto committed:
Merge pull request #18 from Matgenix/flow_info

Fixed jf flow list.

davidwaroquiers committed:
Merge pull request #9 from Matgenix/docs_template

Added documentation structure.

gpetretto committed:
fix flow query for ONGOING state

davidwaroquiers committed:
Added STOPPED and CANCELLED Job States.
Added STOPPED Flow State.
Added proper queries for jobs and flows with a STOPPED state.

davidwaroquiers committed:
Fixed flow queries by state for STOPPED and FAILED.

gpetretto committed:
Merge pull request #22 from Matgenix/stopped_cancelled_states

Stopped states.

gpetretto committed:
jf flow info and other cli updates

gpetretto committed:
fix job query

gpetretto committed:
modify name search field

gpetretto committed:
Merge pull request #23 from Matgenix/flow_info

jf flow info and other cli updates

gpetretto committed:
fix query with aggregation

gpetretto committed:
fix query by name

gpetretto committed:
more filtering options for CLI

gpetretto committed:
tune start date search in CLI

Commit:
fix typo

gpetretto committed:
Merge pull request #25 from FabiPi3/fix_typo

fix typo in cli help message

ml-evs committed:
Initial work on integration tests

Add docker definition for slurm+mongodb+ssh test container

Add initial slurm tests

Add fireworks setup inside docker

Simple flow submission test and daemon teardown

Separate out slurm and database containers

First working version of e2e tests

Remove py310-ism

Tweak dummy version test so it also works locally

Add slightly more complicated test

Only copy src files into docker container to avoid tests invalidating docker cache

Separate out integration and unit tests until isolation can be fixed

Simplify initial tests

Add test from jobflow repo to check fireworks directly

Add test for local shell job

Protect random port selection against flaky docker port authorisation failures

Add 'real' use case tests that currently fail

Replace micromamba with native Python venv

Skip integration tests during unit test run

Remove fireworks from testing set up

Bump docker-sdk

Re-add __init__.py

ml-evs committed:
Run CI on PRs to develop branch

gpetretto committed:
set full path for queue files

davidwaroquiers committed:
Merge pull request #28 from ml-evs/ml-evs/tweak_CI

Run CI on PRs to develop branch

gpetretto committed:
pydantic2 updates

ml-evs committed:
Use the qtoolkit and jobflow PyPI packages, add direct pydantic dep (#31)

* Use the qtoolkit and jobflow PyPI package

* Add direct dependency on pydantic

ml-evs committed:
Fix CI trigger config

ml-evs committed:
Linting fixes and CI config for multiple Python 3.9+ (#34)

* Disable fail-fast

* Disable py38 build

* Remove docs workflow and re-add 3.8 to test matrix

* Linting fixes; set default language version to 3.9 in pre-commit

* Use 3.9 as the default CI version

* Update dummy version tag in tests

* Cancel running CI jobs if new changes are pushed to a PR

* Run pre-commit autoupdate and unpin flake8 plugins

* Add a second dummy test that checks top-level imports

* Drop 3.8 support

* Run linting with new black rules and fix warnings stacklevel warning

* Reinforce python3.9 in pre-commit to avoid linting slosh

* Use Python 3.9 compatible type hints for pydantic objects

gpetretto committed:
add index to the folder name

gpetretto committed:
Merge branch 'develop' into pydantic2

gpetretto committed:
Merge branch 'develop' of github.com:Matgenix/jobflow-remote into pydantic2

gpetretto committed:
fix settings for python 3.9

gpetretto committed:
Merge pull request #35 from Matgenix/fix_index

Add index to the folder name

gpetretto committed:
New queue handling

gpetretto committed:
Merge branch 'develop' into queue

gpetretto committed:
Merge remote-tracking branch 'origin/develop' into queue

gpetretto committed:
Merge branch 'pydantic2' into queue

gpetretto committed:
fix retry

gpetretto committed:
fix py3.9

gpetretto committed:
split runner execution

gpetretto committed:
implement max_jobs per worker. switch to schedule package. Fix for split and pydantic2

gpetretto committed:
first implementation of batch submission

gpetretto committed:
fixes and updates

gpetretto committed:
fix export typing

gpetretto committed:
fix dynamic flow. Add graph utils

gpetretto committed:
some fix and test dash graph app

gpetretto committed:
fix dynamics flow and more graphs

gpetretto committed:
fix cli bug and add docstrings

gpetretto committed:
Avoid Job deserialization in Runner.

gpetretto committed:
fix date inizialization

gpetretto committed:
Merge branch 'queue' into references

gpetretto committed:
fix db_id in output's metadata

gpetretto committed:
fix output

gpetretto committed:
more docstrings

gpetretto committed:
Documentation initial version

davidwaroquiers committed:
Merge pull request #37 from Matgenix/queue

New job management system

davidwaroquiers committed:
Added github release workflow.

gpetretto committed:
Merge branch 'references' into queue

gpetretto committed:
breaking: change db_id type to str

gpetretto committed:
breaking: rename CANCELLED JobState to USER_STOPPED and the related methods accordingly

gpetretto committed:
more documentation

gpetretto committed:
Merge pull request #47 from Matgenix/queue

Breaking changes: db_id type, cancelled state, deserialization and documentation

ml-evs committed:
Merge branch 'develop' into release_workflow

ml-evs committed:
Add docs build to testing workflow

ml-evs committed:
Move docs workflow earlier in release and configure git

ml-evs committed:
Remove unecessary sphinx extension

ml-evs committed:
Do not error on sphinx warnings

ml-evs committed:
Remove defunct docs file

ml-evs committed:
Tidy up empty docs pages and restructure slightly

ml-evs committed:
Fix links in contributing and install docs

ml-evs committed:
Empty the changelog ready for first release

ml-evs committed:
Remove defunct extensions

ml-evs committed:
Auto generate changelog and contributing instructions from markdown with pandoc

ml-evs committed:
Update README links

ml-evs committed:
Add name to author list

ml-evs committed:
Merge pull request #46 from Matgenix/release_workflow

Added GitHub release workflow and docs builds

ml-evs committed:
Fix README badge (#48)

ml-evs committed:
Migrating to newest jfr

Reintroduce tick option for programmatic use of runner

Formatting output

Update tests for latest jfr

ml-evs committed:
Enhance test cases and streamline docker image

Add a test with expected job failures

Fix pre-run config so that slurm jobs actually succeed

Add descriptive comment to dockerfile

Unify test cases for remote/local workers in case of failure

Simplify port selection for Slurm/MongoDB containers

Add `__pycache__` to .dockerignore to improve docker build caching

Update project check test to actually check outputs

Remove versioningit from docker install to improve build caching/size

Separate project check asserts

Tweak pytest invocation

ml-evs committed:
Add codecov upload (#49)

ml-evs committed:
More in-depth test cases

- Add simple SSH test and expected error in case of job with no output

- Give failing test more ticks

ml-evs committed:
Skip integration tests unless the `CI` env var is set

ml-evs committed:
Monkeypatch `fabric.Connection.run` to play nicely with pytest and add
default SSH settings

- Try capturing pytest output again

ml-evs committed:
Pre-commit autoupdate

ml-evs committed:
Bump pyupgrade to 3.9 plus and make fixes

ml-evs committed:
Use recommended pydantic+mypy settings and fix linting

ml-evs committed:
Remove errant `__init__.py` that prevents mypy from working

ml-evs committed:
Pin versions and add requirements lock file for testing and monitoring

ml-evs committed:
Enable dependabot

ml-evs committed:
Add additional sleep before container readiness check

ml-evs committed:
Add test for jobs with callable kwargs (#51)

ml-evs committed:
Add a test that runs with an exec_config and checks an environment variable is set in the job (#56)

ml-evs committed:
Add dev setup docs (#57)

ml-evs committed:
Fix typing of enum values sort and log_level

ml-evs committed:
Fix typing for qresources

ml-evs committed:
File format enum typing

ml-evs committed:
Add missing type hint

ml-evs committed:
Fix pedantic issue with int assignment

ml-evs committed:
Add test for additional stores with GridFS

- Manually connect to gridfs and check for file

- Read from GridFS store directly

- Fix typo

gpetretto committed:
fix daemon configuration with single process

dependabot[bot] committed:
Bump actions/setup-python from 4 to 5

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 5.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/upload-pages-artifact from 2 to 3

Bumps [actions/upload-pages-artifact](https://github.com/actions/upload-pages-artifact) from 2 to 3.
- [Release notes](https://github.com/actions/upload-pages-artifact/releases)
- [Commits](https://github.com/actions/upload-pages-artifact/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/upload-pages-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump codecov/codecov-action from 3 to 4

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3 to 4.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/deploy-pages from 2 to 4

Bumps [actions/deploy-pages](https://github.com/actions/deploy-pages) from 2 to 4.
- [Release notes](https://github.com/actions/deploy-pages/releases)
- [Commits](https://github.com/actions/deploy-pages/compare/v2...v4)

---
updated-dependencies:
- dependency-name: actions/deploy-pages
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pydantic from 2.5.3 to 2.6.1

Bumps [pydantic](https://github.com/pydantic/pydantic) from 2.5.3 to 2.6.1.
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.5.3...v2.6.1)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update pytest requirement from ~=7.2 to ~=8.0

Updates the requirements on [pytest](https://github.com/pytest-dev/pytest) to permit the latest version.
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/7.2.0...8.0.0)

---
updated-dependencies:
- dependency-name: pytest
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

ml-evs committed:
Fix README/pyproject links and add JOSS citation

ml-evs committed:
Update release workflow to commit changelog and move tag

ml-evs committed:
Add lower pin for qtoolkit

matgenix-bot committed:
Release v0.1.0

ml-evs commented:
Now we have released, perhaps this PR can be closed @davidwaroquiers? Or is the plan to sync `main` with develop at some point?

gpetretto committed:
Allow interactive authentication for workers

gpetretto committed:
handle errors in get_pubkeys

ml-evs committed:
Add test for case where additional store is not known to jobflow-remote

ml-evs committed:
Add ability to detect missing configuration of additional stores

- Use inspect instead of wrapped annotations, and return `RemoteError` when store is not present

- Tweak store tests

- Fix bad auto editor import

- Dial back the size of test data in the GridFS test

ml-evs committed:
Reduce pydantic pin to allow installation of atomate2[strict]==0.1.3 (#77)

gpetretto committed:
fix project check for local worker

gpetretto committed:
handle multiple nested gateways in config

gpetretto committed:
minimal version of the JSONStore for internal data transfer

gpetretto committed:
update docs

gpetretto committed:
updates and tests

gpetretto committed:
option to customize remote jobstore

gpetretto committed:
temporarily use the old suuid locally to avoid depreacation warnings

gpetretto committed:
remove redundant operation in Runner

gpetretto committed:
fix job query based on metadata

gpetretto committed:
more fix for job and flow queries

gpetretto committed:
improve docs for multiple projects

ml-evs committed:
Type hints, guards and explicit exports to please mypy

gpetretto committed:
fix queue-out command for interactive login

ml-evs committed:
Partially revert enum changes for typer compat: add mypy ignores

ml-evs committed:
Use TypeAlias to avoid typer's PEP 604 incompatibility

ml-evs committed:
Pass deserialized resources instead of string

gpetretto committed:
Apply suggestions from code review

Co-authored-by: Matthew Evans <7916000+ml-evs@users.noreply.github.com>

ml-evs committed:
Make sure CLI uses type aliases and update resources

gpetretto committed:
Update src/jobflow_remote/testing/__init__.py

Co-authored-by: Matthew Evans <7916000+ml-evs@users.noreply.github.com>

gpetretto committed:
updates after review

ml-evs committed:
Try to remove 'unused' mypy warnings

gpetretto committed:
revert max port in tests

gpetretto committed:
fix linting

gpetretto committed:
fix linting by changing Enum types for typer options

gpetretto committed:
fix lint

gpetretto committed:
handle differently the case of ssh key+OTP

gpetretto committed:
tests and updates

gpetretto committed:
fix tests

gpetretto committed:
fix wrong merge

gpetretto committed:
improve documentation for OTP connection

gpetretto committed:
fix test after rebase

gpetretto committed:
improve CLI project check

gpetretto committed:
adapt integration test

gpetretto committed:
Project docs

dependabot[bot] committed:
Bump qtoolkit from 0.1.3 to 0.1.4

Bumps [qtoolkit](https://github.com/matgenix/qtoolkit) from 0.1.3 to 0.1.4.
- [Release notes](https://github.com/matgenix/qtoolkit/releases)
- [Changelog](https://github.com/Matgenix/qtoolkit/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/matgenix/qtoolkit/compare/v0.1.3...v0.1.4)

---
updated-dependencies:
- dependency-name: qtoolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pydantic from 2.6.1 to 2.6.4

Bumps [pydantic](https://github.com/pydantic/pydantic) from 2.6.1 to 2.6.4.
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.6.1...v2.6.4)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump rich from 13.7.0 to 13.7.1

Bumps [rich](https://github.com/Textualize/rich) from 13.7.0 to 13.7.1.
- [Release notes](https://github.com/Textualize/rich/releases)
- [Changelog](https://github.com/Textualize/rich/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Textualize/rich/compare/v13.7.0...v13.7.1)

---
updated-dependencies:
- dependency-name: rich
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump tomlkit from 0.12.3 to 0.12.4

Bumps [tomlkit](https://github.com/sdispater/tomlkit) from 0.12.3 to 0.12.4.
- [Release notes](https://github.com/sdispater/tomlkit/releases)
- [Changelog](https://github.com/sdispater/tomlkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sdispater/tomlkit/compare/0.12.3...0.12.4)

---
updated-dependencies:
- dependency-name: tomlkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

gpetretto committed:
bump qtoolkit in pyproject.toml

matgenix-bot committed:
Release v0.1.1

gpetretto committed:
minor fix to docs and CLI

gpetretto committed:
add checks for job checkout and improve flow graph

dependabot[bot] committed:
Bump flufl-lock from 8.0.2 to 8.1.0

---
updated-dependencies:
- dependency-name: flufl-lock
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

gpetretto committed:
fix runner bug

dependabot[bot] committed:
Bump typer from 0.9.0 to 0.12.3

Bumps [typer](https://github.com/tiangolo/typer) from 0.9.0 to 0.12.3.
- [Release notes](https://github.com/tiangolo/typer/releases)
- [Changelog](https://github.com/tiangolo/typer/blob/master/docs/release-notes.md)
- [Commits](https://github.com/tiangolo/typer/compare/0.9.0...0.12.3)

---
updated-dependencies:
- dependency-name: typer
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump tomlkit from 0.12.4 to 0.12.5

Bumps [tomlkit](https://github.com/sdispater/tomlkit) from 0.12.4 to 0.12.5.
- [Release notes](https://github.com/sdispater/tomlkit/releases)
- [Changelog](https://github.com/python-poetry/tomlkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sdispater/tomlkit/compare/0.12.4...0.12.5)

---
updated-dependencies:
- dependency-name: tomlkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pydantic from 2.6.4 to 2.7.2

Bumps [pydantic](https://github.com/pydantic/pydantic) from 2.6.4 to 2.7.2.
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.6.4...v2.7.2)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

janosh committed:
list known projects in ProjectUndefined err msg

janosh committed:
del needless dict.keys()

janosh committed:
cls.__name__ -> type(cls)

gpetretto committed:
Handle more states in jobcontroller commands

janosh committed:
migrate linting from flake8, isort, autoflake to ruff (same as atomate2, jobflow now)

janosh committed:
ruff auto fixes

gpetretto committed:
Fix issue n flow state when rerunning jobs

dependabot[bot] committed:
Bump schedule from 1.2.1 to 1.2.2

Bumps [schedule](https://github.com/dbader/schedule) from 1.2.1 to 1.2.2.
- [Changelog](https://github.com/dbader/schedule/blob/master/HISTORY.rst)
- [Commits](https://github.com/dbader/schedule/compare/1.2.1...1.2.2)

---
updated-dependencies:
- dependency-name: schedule
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

janosh committed:
ruff ignore legacy violations

janosh committed:
fix ruff format

gpetretto committed:
make JobController.set_job_doc_properties public

gpetretto committed:
Add --verbose and --outputs options to jf flow delete

gpetretto committed:
rename fixture

dependabot[bot] committed:
Bump psutil from 5.9.8 to 6.0.0

Bumps [psutil](https://github.com/giampaolo/psutil) from 5.9.8 to 6.0.0.
- [Changelog](https://github.com/giampaolo/psutil/blob/master/HISTORY.rst)
- [Commits](https://github.com/giampaolo/psutil/compare/release-5.9.8...release-6.0.0)

---
updated-dependencies:
- dependency-name: psutil
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

matgenix-bot committed:
Release v0.1.2

janosh committed:
Fix legacy `ruff` errors and enable corresponding rules for future linting (#134)

* fix ruff format

* manual ruff fixes, addressing all RET errors

* fix or ignore mypy errors

* fix tests

* fix typos and mypy

* fix linting

* revert print removal

* only ignore subset of pylint refactor rules

* fix legacy bug mentioned in https://github.com/Matgenix/jobflow-remote/pull/134#discussion_r1658643488

QuantumChemist committed:
Adding hyperlink for slurm and pbs resource keywords in tuning.rst

Also see https://github.com/Matgenix/jobflow-remote/issues/143#issuecomment-2223981082

QuantumChemist committed:
fix linting

davidwaroquiers committed:
Fixed yaml issue with deprecated dump function.

davidwaroquiers committed:
Updated version dep for ruamel.yaml in pyproject.toml.

davidwaroquiers committed:
Fixed YAML().

davidwaroquiers committed:
Explicitly setting default_flow_style to False.
